src/GraphQL.Client/GraphQLHttpClientOptions.cs

```diff
@@ -56,5 +56,11 @@ public class GraphQLHttpClientOptions
         /// Configure additional websocket options (i.e. headers). This will not be invoked on Windows 7 when targeting .NET Framework 4.x. 
         /// </summary>
         public Action<ClientWebSocketOptions> ConfigureWebsocketOptions { get; set; } = options => { };
+
+        /// <summary>
+        /// Sets the `ConnectionParams` object sent with the GQL_CONNECTION_INIT message on establishing a GraphQL websocket connection.
+        /// See https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md#gql_connection_init.
+        /// </summary>
+        public Func<object?> SetWebSocketConnectionInitPayload { get; set; } = () => null;
```

Comment:
1. ConfigureWebSocketConnectionInitPayload ?
2. Maybe to pass some context info (at least the id of the request) into delegate - `Func<SomeContext, object?>ConfigureWebSocketConnectionInitPayload {get;set;} = _ => null; ` A separate class will allow you to expand the context in the future while maintaining backward compatibility with the delegate signature.

Reply:
We won't have the request-id in the future (the current implementation of the handshake is flawed, see #293). After #293 is fixed this will only be sent once on connecting the websocket, not for each subscription...

I think this way it's pretty similar to the javascript implementation...

Reply:
I khow that this should be sent only once. Additional context won't hurt I think.

Reply:
But what context? I suspect you don't even need an id for the `GQL_CONNECTION_INIT` request... Everything else is known from the options class..

Reply:
Then at least these options should be passed.